native/src/main/java/io/ballerina/stdlib/mqtt/utils/MqttUtils.java

```diff
@@ -62,20 +67,81 @@
 import static io.ballerina.stdlib.mqtt.utils.MqttConstants.KEY_STORE_PASSWORD;
 import static io.ballerina.stdlib.mqtt.utils.MqttConstants.KEY_STORE_PATH;
 import static io.ballerina.stdlib.mqtt.utils.MqttConstants.MAX_RECONNECT_DELAY;
+import static io.ballerina.stdlib.mqtt.utils.MqttConstants.MESSAGE_ID;
+import static io.ballerina.stdlib.mqtt.utils.MqttConstants.MESSAGE_PROPERTIES;
 import static io.ballerina.stdlib.mqtt.utils.MqttConstants.NATIVE_DATA_PRIVATE_KEY;
 import static io.ballerina.stdlib.mqtt.utils.MqttConstants.NATIVE_DATA_PUBLIC_KEY_CERTIFICATE;
 import static io.ballerina.stdlib.mqtt.utils.MqttConstants.PASSWORD;
 import static io.ballerina.stdlib.mqtt.utils.MqttConstants.PROTOCOL_NAME;
 import static io.ballerina.stdlib.mqtt.utils.MqttConstants.PROTOCOL_VERSION;
+import static io.ballerina.stdlib.mqtt.utils.MqttConstants.REASON_CODE;
+import static io.ballerina.stdlib.mqtt.utils.MqttConstants.RECORD_DELIVERY_TOKEN;
+import static io.ballerina.stdlib.mqtt.utils.MqttConstants.RESPONSE_TOPIC;
 import static io.ballerina.stdlib.mqtt.utils.MqttConstants.SECURE_SOCKET;
 import static io.ballerina.stdlib.mqtt.utils.MqttConstants.SERVER_URIS;
+import static io.ballerina.stdlib.mqtt.utils.MqttConstants.TOPIC;
 import static io.ballerina.stdlib.mqtt.utils.MqttConstants.USERNAME;
 
 /**
  * Class containing the utility functions related to the clients.
  */
 public class MqttUtils {
 
+    public static BMap<BString, Object> getBMqttMessage(MqttMessage message, String topic) {
+        BMap<BString, Object> bMessage = ValueCreator.createRecordValue(ModuleUtils.getModule(),
+                MqttConstants.RECORD_MESSAGE);
+        bMessage.put(StringUtils.fromString(MqttConstants.PAYLOAD),
+                ValueCreator.createArrayValue(message.getPayload()));
+        bMessage.put(StringUtils.fromString(MqttConstants.MESSAGE_ID), message.getId());
+        bMessage.put(StringUtils.fromString(MqttConstants.QOS), message.getQos());
+        bMessage.put(StringUtils.fromString(MqttConstants.RETAINED), message.isRetained());
+        bMessage.put(StringUtils.fromString(MqttConstants.DUPLICATE), message.isDuplicate());
+        bMessage.put(MqttConstants.TOPIC, StringUtils.fromString(topic));
+        MqttProperties properties = message.getProperties();
+        if (properties != null) {
```

Comment:
Lets use `Objects.nonNull` here as well (for consistency across the code-base)

Reply:
Fixed in https://github.com/ballerina-platform/module-ballerina-mqtt/pull/9/commits/041d5b9016d02da09f39901fe4d4970544edfb25